### PR TITLE
feat: Improve Athena Dump reader

### PR DESCRIPTION
**Pull request by benjaminhuth**
Can read more things now.

### CODE REVIEW

File: Examples/Io/Root/src/RootAthenaDumpReader.cpp [hunk 1]
```diff
@@ -265,11 +340,46 @@ ActsExamples::ProcessCode ActsExamples::RootAthenaDumpReader::read(
...
+    DigitizedParameters digiPars;
+    if (type == ePixel) {
+      digiPars.indices = {Acts::eBoundLoc0, Acts::eBoundLoc1};
+      digiPars.values = {CLloc_direction1[im], CLloc_direction2[im]};
+      assert(locCov.size() == 4);
+      digiPars.variances = {locCov[0], locCov[3]};
+    } else {
+      // TODO is this correct ???
+      digiPars.values = {CLloc_direction2[im]};
+      digiPars.indices = {Acts::eBoundLoc1};
```
Comment (andiwand) [inline Examples/Io/Root/src/RootAthenaDumpReader.cpp:366]: is it? 😄

File: Examples/Io/Root/src/RootAthenaDumpReader.cpp [hunk 2]
```diff
@@ -265,11 +340,46 @@ ActsExamples::ProcessCode ActsExamples::RootAthenaDumpReader::read(
...
+
+      // why is this 4
+      // assert(locCov.size() == 1);
+      digiPars.variances = {locCov[0]};
```
Comment (andiwand) [inline Examples/Io/Root/src/RootAthenaDumpReader.cpp:370]: can the assert be used for testing or always or should it be dropped?

File: Examples/Io/Root/src/RootAthenaDumpReader.cpp [hunk 3]
```diff
@@ -14,10 +14,60 @@
...
+
+#include <cmath>
 
 #include <TChain.h>
 #include <boost/container/static_vector.hpp>
 
+class BarcodeConstructor {
+  /// Particles with barcodes larger then this value are considered to be
+  /// secondary particles
+  constexpr static int s_maxBarcodeForPrimary = 200000;
```
Comment (andiwand) [inline Examples/Io/Root/src/RootAthenaDumpReader.cpp:27]: does this magic number have an origin or is it an arbitrary high number?

**Review by andiwand — DISMISSED** [review]
LGTM 👍 left a few minor comments

File: Examples/Io/Root/src/RootAthenaDumpReader.cpp (hunk 3 above)
Comment (benjaminhuth) [reply Examples/Io/Root/src/RootAthenaDumpReader.cpp:27]: Hmm I can link to Athena code where it comes from...

File: Examples/Io/Root/src/RootAthenaDumpReader.cpp (hunk 1 above)
Comment (benjaminhuth) [reply Examples/Io/Root/src/RootAthenaDumpReader.cpp:366]: Ah yeah good that you mention, I should check before merging^^

**Review by andiwand — APPROVED** [review]

**Comment by paulgessinger** [comment]
Is this ready or still blocked?

**Comment by benjaminhuth** [comment]
Sorry forgot the label

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [6334171f3c86a565a9fb28076b53cae98f779d15]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/41122704) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7738809)!

**Pull request merged**